src/storage-types/src/controller.rs

```diff
@@ -256,7 +257,9 @@ impl fmt::Display for StorageError {
                 write!(
                     f,
                     "expected upper was different from the actual upper for: {}",
-                    id.iter().map(|id| id.to_string()).join(", ")
+                    id.iter()
+                        .map(|invalid_upper| format!("{invalid_upper:?}"))
```

Comment:
`Antichain`s don't implement `Display` (as far as I could tell) so this is the best I could come up with.

Reply:
There is also a `.pretty()` method on `Antichain<T: Display>` with the extension trait in `mz_timely_util` which will display it as `{42}`

Reply:
Just updated to use `.pretty()`.